src/handleLocalStorage.ts

```diff
@@ -0,0 +1,29 @@
+import { EventData } from "./types/main.js";
+
+const currentEventDataKey: string = "currentEventData";
+const savedEventsKey: string = "allSavedEvents";
+
+const storeDataInLocalStorage = (key: string, data: EventData | EventData[]) => {
+    try {
+        if (key && data !== undefined) {
```

Comment:
data will never be undefined because your type states that it is EventData or array of EventData
```suggestion
        if (key && data !== undefined) {
```